### ISSUE TITLE
[Bug]: Forgot Password | Auth API error: bad_jwt

### ISSUE THREAD

**Issue by Shabinder** [supabase-community/supabase-kt#706]
### General Info

- [X] I checked for similar bug report
- [X] I am using the latest version
- [X] I checked the [troubleshooting](https://github.com/supabase-community/supabase-kt/blob/master/TROUBLESHOOTING.md) page for similar problems

### Version(s)

2.6.0

### Kotlin Target(s) and their respective versions

JVM 21

### What happened? (include your code)

Refer Flow below:

 Note: using PKCE auth flow.

### Steps To Reproduce (optional)

1.  Create an email account.
2.  Logout.
3.  Forgot Password
4.  Received a Deep link on Mail
5.  Clicking it, Opens App, And logs in.
6.  App Prompts for a new password.
7.  User enters a new password, and we call `auth.updateUser { this.password = newPassword }`
8.  Request Fails with `Auth API error: bad_jwt`

 

### Relevant log output (optional)
```
io.github.jan.supabase.gotrue.exception.AuthRestException: invalid JWT: unable to parse or verify signature, token signature is invalid: signing method RS256 is invalid
	at io.github.jan.supabase.gotrue.AuthImpl.checkErrorCodes(AuthImpl.kt:523)
	at io.github.jan.supabase.gotrue.AuthImpl.parseErrorResponse(AuthImpl.kt:491)
	at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApiKt$authenticatedSupabaseApi$3.invoke(AuthenticatedSupabaseApi.kt:60)
	at io.github.jan.supabase.gotrue.AuthenticatedSupabaseApiKt$authenticatedSupabaseApi$3.invoke(AuthenticatedSupabaseApi.kt:60)
	at io.github.jan.supabase.network.SupabaseApi.rawRequest$suspendImpl(SupabaseApi.kt:25)
	at io.github.jan.supabase.network.SupabaseApi$rawRequest$1.invokeSuspend(SupabaseApi.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:101)
	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:113)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:89)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:589)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:823)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:720)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:707)
```

**Comment by jan-tennert** [comment]
Seems like you are having the same problem as in #704. This doesn't looks like it's supabase-kt's fault (just tried a similar forgot password flow)
Are you using a self-hosted Supabase instance?

**Comment by Shabinder** [comment]
> Are you using a self-hosted Supabase instance?
Yes, I am.

**Comment by J0** [comment]
Hey team,

Thanks for flagging the issue.

The above error typically indicates that there’s a signing mismatch (e.g. the token was signed symmetrically and the received token is asymmetric or the secret the token was signed with doesn’t tally with the signature component of the token)

Could we trouble you to share the non sensitive components of the .env config for the Auth component of the self-hosted stack if available? That would help

Let us know

**Comment by Shabinder** [comment]
Yes, I can share the needed info.

Also an update.
since I am not in production, so I experimented, by creating a supabase test project and then I setup auth with providers and redirect uris there, and used the client with that, all flows worked.

So, then I used the JWT, Anon, Secrets from the supabase test project to setup a new docker environmet, so that I could conclude there is no other difference except being a hosted & cloud version, and the issue still persists.

**tldr: using same secrets as in cloud test project in my hosted version,
Client works with cloud url but fails with above trace in hosted version.**

<details>
  <summary>My .env</summary>

```
SITE_URL=https://mysite.in
ADDITIONAL_REDIRECT_URLS=xxxx,xxxx,xxxs
JWT_EXPIRY=3600
DISABLE_SIGNUP=false
API_EXTERNAL_URL=https://mysite.com

## Mailer Config
MAILER_URLPATHS_CONFIRMATION="/auth/v1/verify"
MAILER_URLPATHS_INVITE="/auth/v1/verify"
MAILER_URLPATHS_RECOVERY="/auth/v1/verify"
MAILER_URLPATHS_EMAIL_CHANGE="/auth/v1/verify"

ENABLE_ANONYMOUS_USERS=false

## Email auth
ENABLE_EMAIL_SIGNUP=true
ENABLE_EMAIL_AUTOCONFIRM=true # passthrough.
SMTP_ADMIN_EMAIL=my@mysite.com
SMTP_HOST=smtp.host.com
SMTP_PORT=587
SMTP_USER=user@site.com
SMTP_PASS=pass
SMTP_SENDER_NAME=MyName
EMAIL_SUBJECT="Please confirm your email"

## Phone auth
ENABLE_PHONE_SIGNUP=true
ENABLE_PHONE_AUTOCONFIRM=true

## Google Auth
ENABLE_GOOGLE_SIGNUP=true
GOOGLE_CLIENT_ID=xxx
GOOGLE_CLIENT_SECRET=xxx
```

</details

**Comment by Shabinder** [comment]
lmk if I can share/do anything more to help, 
will be blocked on this..., will wait to hear back from you.

**Comment by jan-tennert** [comment]
We'll probably have to wait until J0 responds, but I could imagine that re-using the secrets from the hosted version may be a problem.

**Comment by Shabinder** [comment]
> We'll probably have to wait until J0 responds

Yeah, will be doing that, since can't move forward with finalising supabase as the backend of choice, until this blocker is cleared.

**Comment by Shabinder** [comment]
More Diagnosing Data:
I verified my session token which is being sent, see below image, its correctly using HS256,

<img width="1136" alt="image" src="https://github.com/user-attachments/assets/cfd3dde8-eb1d-4c86-854d-df998e2b3c40">

So the issue now is,  Why supabase/auth is responding back an error saying  algo is RS256...

**Comment by Shabinder** [comment]
I am back 🙃 ,

after trying a lot of things, I found, that performing same curl from localhost directly on port, worked,

then tested my nginx config, and found it was quite alright, had some issue in my cloudflare setup of domain forwarding.

After fixing that, all works like a charm.

I hope this whole errand, might be helpful for someone who faces same at a later point and he may remember checking all above first.

Nonetheless, thanks guys, I appreciate all the help.
